src/main/java/com/vmware/vipclient/i18n/messages/service/FormattingCacheService.java

```diff
@@ -20,56 +20,58 @@ public FormattingCacheService() {
 
     }
 
-    public void addPatterns(String locale, JSONObject o) {
+    public void addPatterns(String locale, PatternCacheItem o) {
         String cacheKey = getPatternsCacheKey(locale);
         addFormattings(cacheKey, o);
     }
 
-    public void addPatterns(String language, String region, JSONObject o) {
+    public void addPatterns(String language, String region, PatternCacheItem o) {
         String cacheKey = getPatternsCacheKey(language, region);
         addFormattings(cacheKey, o);
     }
 
-    public void addSupportedLanguages(BaseDTO dto, String locale, Map<String, String> o) {
-        o = JSONUtils.map2SortMap(o);
+    public void addSupportedLanguages(BaseDTO dto, String locale, LocaleCacheItem o) {
+        Map<String, String> map = JSONUtils.map2SortMap(o.getCachedData());
+        o.addCachedData(map);
```

Comment:
Remove addCachedData setter method. Set cachedData together with the other properties in a single synchronized setter method. See my comment in RemoteLocaleOpt.java line 40

Do this sorting inside that setter, right before setting the cacheItem's cachedData.